### PR TITLE
ESA Pull Request

### PR DESCRIPTION
**Pull request by deusbajaj**

### CODE REVIEW

File: R/ESA.R [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# Input: The biological system PHALY and a gene the user picks from the system
+# Output: A histogram for the node degree distribution of the PHALY gene network
+# and a network plot for a gene from the PHALY system
+# Dependencies: devtools, iGraph
+#
+
+# ====  FUNCTIONS  =============================================================
+
+# function stub taken from Dr. Steipe's BCB420.2019.ESA package (https://github.com/hyginn/BCB420.2019.ESA)
+fetchComponents <- function(sys) {
```
Comment (rn1n3r) [inline R/ESA.R]: You don't need this function stub, and fetchComponents has been deprecated anyways. Please use `SyDBgetSysSymbols(myDB, sys)` instead

File: R/ESA.R [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+#' \code{systemAnalysis()} Investigates genes from the system PHALY
+#' and produces plots with gene network data
+#'
+#' @param gene A gene from the Biological system PHALY
+#' @return a histogram for the node degree distribution of the PHALY gene network
+#' and a network plot for a gene from the PHALY system
+#'
+#' @author {Deus Bajaj} (aut)
+#'
+#' @examples
```
Comment (rn1n3r) [inline R/ESA.R:29]: Please leave a short comment explaining what your code sample is doing.

File: R/ESA.R [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+systemAnalysis <- function(gene) {
+
+  # Import STRINGedges from git repo at
+  # https://github.com/deusbajaj/BCB420.2019.ESA
+  # created from Dr. Steipe's BCB420.2019.ESA package
+  # (forked directory, master repo can be accessed at
+  # https://github.com/hyginn/BCB420.2019.ESA)
+
+  # Load STRINGedges
+  myURL <- paste0("http://steipe.biochemistry.utoronto.ca/abc/assets/",
```
Comment (rn1n3r) [inline R/ESA.R]: To load STRING data, you should use the included `fetchData()` function instead.

File: R/ESA.R [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+#' and a network plot for a gene from the PHALY system
+#'
+#' @author {Deus Bajaj} (aut)
+#'
+#' @examples
+#' systemAnalysis("AMBRA1")
+#'
+#' @export
+
+systemAnalysis <- function(gene) {
```
Comment (rn1n3r) [inline R/ESA.R]: Just a suggestion but perhaps you could take both the gene and system as parameters so your function isn't restricted to just analysis of the PHALY system

File: R/ESA.R [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  }
+  return(s)
+}
+
+#'
+#' \code{systemAnalysis()} Investigates genes from the system PHALY
+#' and produces plots with gene network data
+#'
+#' @param gene A gene from the Biological system PHALY
+#' @return a histogram for the node degree distribution of the PHALY gene network
```
Comment (rn1n3r) [inline R/ESA.R]: This function plots a histogram but doesn't actually return anything. Consider returning the graph or maybe the degree distribution so the user can conduct further analysis if necessary.

**Review by rn1n3r — CHANGES_REQUESTED** [review]

File: R/ESA.R (hunk 4 above)
Comment (deusbajaj) [reply R/ESA.R]: That's actually a really good suggestion, I'll add that in.

**Comment by deusbajaj** [comment]
Fixed all the issues based on the comments in the review.

**Comment by deusbajaj** [comment]
The last commit is not triggering the Travis CI build so will be closing this and creating a new pull request.

**Pull request closed without merging**